sdk/tests/core/CMakeLists.txt

```diff
@@ -18,6 +18,11 @@ endif()
 
 create_map_file(az_core_test.map)
 
+set(MATH_LIB_UNIX "")
+if (UNIX)
+    set(MATH_LIB_UNIX "m")
+endif()
```

Comment:
nit: Add a comment explaining why we add this.

From https://github.com/Azure/azure-sdk-for-c/pull/1377/files#diff-ad1c9b0219da4ff1631f24a31f70d040L60
> add math lib on Linux only

Is this needed specific to our tests only?

Reply:
yes, az_span needs the math lib for `az_span_dtoa` -> `double after_decimal_part = modf(source, &integer_part);`  (modf is at math)